File: src.csharp/AlphaTab/Core/EcmaScript/Array.cs
using System.Collections;
using System.Collections.Generic;
using System.Linq;

namespace AlphaTab.Core.EcmaScript;

internal static class Array
{
    public static bool IsArray(object? o)
    {
        return o is IList;
    }

    public static IList<T> From<T>(IEnumerable<T> x)
    {
        return x.ToList();
    }

    public static IList<T> From<T>(IEnumerator<T> x)
    {
        var list = new List<T>();
        foreach (var i in x)
        {
            list.Add(i);
        }
        return list;
    }
}

File: src.kotlin/alphaTab/shared/src/commonMain/kotlin/alphaTab/core/ecmaScript/Array.kt
package alphaTab.core.ecmaScript

class Array {
    companion object {
        public fun <T> from(x: Iterable<T>): alphaTab.collections.List<T> {
            return alphaTab.collections.List(x)
        }
        public fun isArray(x:Any?):Boolean {
            return x is MutableList<*>
        }
    }
}
